modules/hardware/lenovo-x1/default.nix

```diff
@@ -24,19 +24,24 @@ in {
     ghaf.hardware.definition = {
       inherit (hwDefinition) generic;
       inherit (hwDefinition) input;
-      inherit (hwDefinition) disks;
       inherit (hwDefinition) network;
       inherit (hwDefinition) gpu;
       inherit (hwDefinition) audio;
       inherit (hwDefinition) usb;
     };
 
     # Disk configuration
-    disko.devices.disk = hwDefinition.disks;
+    disko.devices.disk = {
+      disk1.device = "/dev/nvme0n1";
+      disk1.imageSize = "250G";
+    };
 
     # Hardware specific kernel parameters
     boot = {
       inherit (hwDefinition.generic) kernelParams;
+      initrd.availableKernelModules = ["nvme" "zfs"];
+      supportedFilesystems = ["zfs"];
+      kernelPackages = config.boot.zfs.package.latestCompatibleLinuxPackages;
```

Comment:
@vunnyso FYI - this requires follow-up work with the ghaf kernel hardening but is fine in the context of this PR.
In the follow-up work, it might be the easiest to add the ZFS required kernel config fragments to `host_hardened_kernel` https://github.com/tiiuae/ghaf/blob/ae44f5928131415ab578cbe5051baa9c7f78c612/modules/hardware/x86_64-generic/kernel/host/default.nix#L60